sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2493,6 +2493,9 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
    * Note: CTEs are handled in CTESubstitution.
    */
   object ResolveSubquery extends Rule[LogicalPlan] {
+    // This tag is used to mark resolved subquery to avoid resolving it repeatedly.
```

Comment:
Can we add more comments? It's used instead of `plan.resolved` because the post-hoc resolution rules should be run even if the plan is already resolved.